src/modules/iasql_functions/rpcs/iasql_create_review.ts

```diff
@@ -0,0 +1,103 @@
+  RpcBase,
+  RpcResponseObject,
+} from '../../interfaces';
+
+/**
+ * Method that generates creates the markdown to be shared or used in any version control system
+ */
+export class IasqlCreateReview extends RpcBase {
+  /** @internal */
+  module: IasqlFunctions;
+  /** @internal */
+  preTransactionCheck = PreTransactionCheck.NO_CHECK;
+  /** @internal */
+  postTransactionCheck = PostTransactionCheck.NO_CHECK;
+  /** @internal */
+  inputTable: RpcInput = {
+    title: { argType: 'varchar', default: 'New Review' },
+    description: { argType: 'varchar', default: 'Review description' },
+  };
+  /** @internal */
+  outputTable = {
+    markdown: 'text',
+  } as const;
+
+  documentation = {
+    description: 'Generate review markdown for latest transaction',
+    sampleUsage: "SELECT * FROM iasql_create_review('My review title', 'Meaningful description')",
+  };
+
+  /** @internal */
+  call = async (
+    _dbId: string,
+    _dbUser: string,
+    ctx: Context,
+    title: string,
+    description: string,
+  ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
+    let preview: any[];
+    let transactionSql: { sql: string }[];
+    try {
+      preview = (await ctx.orm.query('SELECT * FROM iasql_preview();')) ?? [];
+      // TODO: update once `iasql_get_sql_for_transaction` with no arguments works
+      transactionSql =
+        (await ctx.orm.query(`
+        SELECT *
+        FROM iasql_get_sql_for_transaction(
+          (
+            SELECT transaction_id
+            FROM iasql_audit_log
+            WHERE change_type = 'OPEN_TRANSACTION'
+            ORDER BY ts DESC
+            LIMIT 1
+          )
+        );
+      `)) ?? [];
+    } catch (e) {
+      throw e;
+    }
+    return [
+      {
+        markdown: generateMarkdown(title, description, preview, transactionSql?.[0]?.sql),
+      },
+    ];
+  };
+
+  constructor(module: IasqlFunctions) {
+    super();
+    this.module = module;
+    super.init();
+  }
+}
+
+function generateMarkdown(title: string, description: string, preview: any[], transactionSql?: string) {
+  const titleMarkdown = `# ${title}`;
+  const descriptionMarkdown = description;
+  if (!preview.length) return;
+  const previewTitleMarkdown = '## IaSQL Preview';
+  const previewKeys = Object.keys(preview[0]);
+  const previewHeaderDelimiters = previewKeys.map(() => '---');
+  const previewHeaderMarkdown = `| ${previewKeys.join(' | ')} |`;
+  const previewHeaderDelimitersMarkdown = `| ${previewHeaderDelimiters.join(' | ')} |`;
+  const previewRowsMarkdown = preview.map(row => {
+    const rowValues = previewKeys.map(key => row[key]);
+    return `| ${rowValues.join(' | ')} |`;
+  });
+  const previewMarkdown = [
+    previewTitleMarkdown,
+    previewHeaderMarkdown,
+    previewHeaderDelimitersMarkdown,
+    ...previewRowsMarkdown,
+  ].join('<-- insert line break here -->');
+  const sqlTitleMarkdown = '## SQL changes';
+  const sqlMarkdown = `\`\`\`sql<-- insert line break here -->
```

Comment:
Why not insert the newlines in the output instead of these comments?

There is a minor formatting bug we should fix with the dashboard to properly render newlines in text (by replacing with `<br />` probably) but this manual post-processing step for the user seems unnecessary if we fix that?

Reply:
Oh, I meant:

1. The postgres function outputs newlines like normal (so it also works as expected if you called this via the `psql` cli app)
2. The dashboard, when it renders text, does a `s/\n/<br />/g` on the text before passing it to table so it renders correctly in there and simply copy-pasting the text brings back the newlines.

Reply:
Thanks! Got it working :awesome:

# My review

Review description

## IaSQL Preview

| action | table_name | id | description |
| --- | --- | --- | --- |
| update | log_group | 4 | quickstart-log-group|us-east-2 |

## SQL changes

```sql

UPDATE log_group
SET log_group_name = 'quickstart-log-group', log_group_arn = 'test', creation_time = '2023-01-25T11:17:14.045+00:00', region = (SELECT region FROM aws_regions WHERE region = 'us-east-2')
WHERE log_group_name = 'quickstart-log-group' AND log_group_arn = 'arn:aws:logs:us-east-2:257682470237:log-group:quickstart-log-group:*' AND creation_time = '2023-01-25T11:17:14.045+00:00' AND region = (SELECT region FROM aws_regions WHERE region = 'us-east-2');

```